staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -152,7 +152,7 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 		if len(name) == 0 {
 			_, name, _ = scope.Namer.ObjectName(obj)
 		}
-		if len(namespace) == 0 && *gvk == namespaceGVK {
+		if len(namespace) == 0 && scope.Resource == namespaceGVR {
```

Comment:
this is existing, but comparing to the request scope resource rather than the decoded object Kind is more correct